File: problem2.rb
def fibonacci
  prev_num = 0
  next_num = 1
  interim_num = 0
  sequence = []
  max = 4000000
  while next_num < max do
    sequence.push(next_num)
    interim_num = next_num
    next_num = next_num + prev_num
    prev_num = interim_num
  end
  even_items = []
  for num in sequence
    if num % 2 == 0
      even_items.push(num)
    end
  end
  even_items.inject(0){|sum,x| sum + x}
end

File: problem2.js
function fibonacci () {
  var prev = 0
  var next = 1
  var interim
  var sequence = []
  while (next < 4000000) {
    sequence.push(next)
    interim = next
    next = next + prev
    prev = interim
  }
  return sumEvenItems(sequence)

  function sumEvenItems (arr) {
    var evenItems = []
    for (var i = 0; i < arr.length; i++) {
      if ((arr[i] % 2) == 0) {
        evenItems.push(arr[i])
      }
    }
    return evenItems.reduce(function (sum, value) {
      return sum + value
    })
  }
}
